examplebroker/broker.go

```diff
@@ -39,25 +43,34 @@ const (
 )
 
 const (
-	// AuthGranted is the response when the authentication is granted.
-	AuthGranted = "granted"
-	// AuthDenied is the response when the authentication is denied.
-	AuthDenied = "denied"
-	// AuthCancelled is the response when the authentication is cancelled.
-	AuthCancelled = "cancelled"
-	// AuthRetry is the response when the authentication needs to be retried (another chance).
-	AuthRetry = "retry"
-	// AuthNext is the response when another MFA (including changing password) authentication is necessary.
-	AuthNext = "next"
+	passwordID       = "password"
+	pinCodeID        = "pincode"
+	totpWithButtonID = "totp_with_button"
+	totpID           = "totp"
+	phoneAck1ID      = "phoneack1"
+	phoneAck2ID      = "phoneack2"
+	fidoDevice1ID    = "fidodevice1"
+	qrCodeID         = "qrcodewithtypo"
+	qrCodeAndCodeID  = "qrcodeandcodewithtypo"
+	codeID           = "codewithtypo"
+	webViewID        = "webview"
```

Comment:
Those represent modes, so appending `Mode` should be more descriptive than `ID`:

```suggestion
	webViewMode        = "webview"
```

Reply:
Aren't technically the authentication mode ID that we expose too?

Reply:
You're right, we do currently have the distinction between an authentication mode and an authentication mode name or ID (which are used interchangeably). So with the current code base, `webViewModeID` or `webViewModeName` would be the most descriptive.

I would like to make another suggestion though. Instead of separating the auth mode name/ID from the `authMode` struct, how about we make it a field of that struct:

```diff
diff --git a/examplebroker/broker.go b/examplebroker/broker.go
index 7b3498ac..33625499 100644
--- a/examplebroker/broker.go
+++ b/examplebroker/broker.go
@@ -57,6 +57,7 @@ const (
 )
 
 type authMode struct {
+       name           string
        selectionLabel string
        ui             map[string]string
        email          string
```

Instead of instantiating the different auth modes in `getSupportedModes`, we could define them as global variables (constants would be even better, but Go doesn't support that for structs):

```go
var passwordMode = authMode{
	name:           "password",
	selectionLabel: "Password authentication",
	ui: map[string]string{
		layouts.Type:  layouts.Form,
		layouts.Label: "Gimme your password",
		layouts.Entry: entries.CharsPassword,
	},
}
```

Then in `getSupportedModes`, we can add them to the supported modes like this:

```
if slices.Contains(supportedEntries, entries.CharsPassword) {
    supportedModes[passwordMode.name] = passwordMode
```

and use `passwordMode.name` throughout the code instead of `passwordID`. WDYT? I could also do that in a follow-up.

Reply:
Fair enough, I can handle that as an extra commit, as I didn't want to make the diff too huge for differences.

Reply:
The only case isn't very clean is the email cases where the modes are just templates